rclpy/src/rclpy/_rclpy.c

```diff
@@ -1129,7 +1130,7 @@ _expand_topic_name_with_exceptions(const char * topic, const char * node, const
     if (rcutils_ret == RCUTILS_RET_BAD_ALLOC) {
       PyErr_Format(PyExc_MemoryError, "%s", rcutils_get_error_string().str);
     } else {
-      PyErr_Format(PyExc_RuntimeError, "%s", rcutils_get_error_string().str);
+      PyErr_Format(RCLError, "%s", rcutils_get_error_string().str);
```

Comment:
I think this change should be reverted since the error isn't coming from `rcl`. A RuntimeError is probably fine for now.

```suggestion
      PyErr_Format(PyExc_RuntimeError, "%s", rcutils_get_error_string().str);
```